api/models/channel.js

```diff
@@ -80,23 +80,14 @@ const getChannelsByUserAndCommunity = async (communityId: string, userId: string
 };
 
 const getChannelsByUser = (userId: string): Promise<Array<DBChannel>> => {
-  return (
-    db
-      .table('usersChannels')
-      // get all the user's channels
-      .getAll(userId, { index: 'userId' })
-      // only return channels where the user is a member
-      .filter({ isMember: true })
-      // get the channel objects for each channel
-      .eqJoin('channelId', db.table('channels'))
-      // get rid of unnecessary info from the usersChannels object on the left
-      .without({ left: ['id', 'channelId', 'userId', 'createdAt'] })
-      // zip the tables
-      .zip()
-      // ensure we don't return any deleted channels
-      .filter(channel => db.not(channel.hasFields('deletedAt')))
-      .run()
-  );
+  return db
+    .table('usersChannels')
+    .getAll([userId, 'member'], { index: 'userIdAndRole' })
```

Comment:
We need to fetch channels a user is a moderator and an owner in here too, right?

```suggestion
    .getAll([userId, 'member'], [userId, 'moderator'], [userId, 'owner'], { index: 'userIdAndRole' })
```

Reply:
I guess again, I'm confused by this index because in our database if you are a moderator or an owner you are *also* a member. So shouldn't fetching `[userId, 'member']` always return members including the team?

Reply:
No, it can only have a single value, you're either-or

Reply:
I guess I'm saying won't the db lookup return the proper results regardless?

Reply:
No. The index only has a single value, so you're **either** a "member" **or** a "moderator" **or** an "owner". Members are exclusive of moderators are exclusive of owners, these groups have 0 overlap in the index.